CHANGELOG.md

```diff
@@ -3,14 +3,57 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## v1.0dev - [date]
+## [[1.0.0](https://github.com/sanger-tol/blobtoolkit/releases/tag/1.0.0)] – ??? - [2023-04-06]
 
-Initial release of nf-core/blobtoolkit, created with the [nf-core](https://nf-co.re/) template.
+### Enhancements & fixes
 
-### `Added`
+Initial release of sanger-tol/blobtoolkit :tada:
 
-### `Fixed`
+- Required input: (1) Tree of Life organism ID (ToLID, --input) or (2) FASTA genome file (--fasta) and `samplesheet.csv` (--input) containing paths to aligned reads in CRAM format, see description of _Full samplesheet_ in `docs/usage.md`.
+- Lineages to run BUSCO are obtained using [goat_taxonsearch module](https://nf-co.re/modules/goat_taxonsearch) instead of reading them from `config.yaml` file (--taxon: NCBI taxonomy ID or species binomial name).
+- Versions of tools are collected using [custom_dumpsoftwareversions module](https://nf-co.re/modules/custom_dumpsoftwareversions) instead of reading them from `config.yaml` file.
+- The pipeline requires aligned reads, `minimap2` is no longer included.
+- A GCA accession ID or YAML file is required (--accession or --yaml) to allow compatibility with `blobtoolkit` scripts. These are used to either generate or read a `config.yaml` file but do not pass parameters to the pipeline. Also software versions in this file do not reflect the actual versions used in the pipeline. 
+- [fastawindows module](https://nf-co.re/modules/fastawindows) computes statistics across windows over the genome FASTA file.
+- Parameters for [diamond_blastp module](https://nf-co.re/modules/diamond_blastp) should be specified in `nextflow.config` file.
 
-### `Dependencies`
+This release marks the point where the pipeline was moved from Snakemake at [blobtoolkit/blobtoolkit](https://github.com/blobtoolkit/blobtoolkit) over to Nextflow DSL2 at [sanger-tol/blobtoolkit](https://github.com/sanger-tol/blobtoolkit). There are two subworkflows in the Snakemake version that are still being implemented in Nextflow: `diamond_blastx` and `blastn`.
 
-### `Deprecated`
+### Parameters
+
+| Old parameter | New parameter   |
+| ------------- | --------------- |
+|               | --accession     |
+|               | --blastp_outext |
+|               | --blastp_cols   |
+|               | --fasta         |
+|               | --input         |
+|               | --taxon         |
+|               | --yaml          |
```

Comment:
This table means that *every* parameter of the Snakemake pipeline is still supported and only new options were added, which I find strange.

Reply:
For almost all our other pipelines, there isn't a previous version to compare against and the 1.0 entry of the changelog doesn't need a table "old parameter" vs "new parameter": `usage.md` suffices.
I'm OK doing the same here and skipping the table. However, there should still be somewhere in this file an introduction about BTK Nextflow 1.0, and how there isn't feature parity with the Snakemake version. We should also say:

- it will _never_ reach feature-parity because we're taking the occasion to reimplement things differently, c.f. the inclusion of `fasta_windows`
- the actual list of the BTK analyses implemented

Reply:
> For almost all our other pipelines, there isn't a previous version to compare against and the 1.0 entry of the changelog doesn't need a table "old parameter" vs "new parameter": `usage.md` suffices. I'm OK doing the same here and skipping the table. However, there should still be somewhere in this file an introduction about BTK Nextflow 1.0, and how there isn't feature parity with the Snakemake version. We should also say:
> 
>     * it will _never_ reach feature-parity because we're taking the occasion to reimplement things differently, c.f. the inclusion of `fasta_windows`
> 
>     * the actual list of the BTK analyses implemented

@priyanka-surana @muffato Regarding this explanation about feature parity: should they be included in `usage.md `as well as `CHANGELOG.md`? Or at least a more detailed description on how Nextflow handles the parameters vs the Snakemake pipeline.

Reply:
I feel the best thing is to pretend our users know nothing about the snakemake pipeline, treat this as an independent pipeline. We can acknowledge and explain the relationship in README.md, CHANGELOG.md and CITATIONS.md, as appropriate but not too detailed.

I think both the parameters and the dependency table are useful for record keeping, but instead of comparing to snakemake, just add everything in the new column as this is first release. It will help users to have the record when we do the next release.

Reply:
I don't think we can "pretend our users know nothing about the snakemake pipeline". We know that ToLA, Mark's team, Mara's team, etc, regularly use the Snakemake pipeline, and many of them can be considered advanced users. They need to know clearly, in advance, how the Nextflow version differs from the Snakemake one, otherwise they'll feel duped.

Reply:
In the meantime I added a small description on how to specify parameters/inputs in Nextflow version compared to Snakemake version (also linked in CHANGELOG.md): 

https://github.com/sanger-tol/blobtoolkit/blob/cleanup/docs/usage.md#yaml-file-and-nextflow-configuration

This could be further improved by suggestions from the users as well. A couple of examples on how to change parameters could also be added in the documentation. Maybe if this part of the documentation is clear there is no need for the Parameters table, at least for the first release.

Reply:
> I don't think we can "pretend our users know nothing about the snakemake pipeline". We know that ToLA, Mark's team, Mara's team, etc, regularly use the Snakemake pipeline, and many of them can be considered advanced users. They need to know clearly, in advance, how the Nextflow version differs from the Snakemake one, otherwise they'll feel duped.

None of those users will be duped as they are aware of our work, plus duping people or taking undue credit has never been the intention. But I still prefer ease of understanding and a degree of abstraction in the documentation instead of over complicating with too much details. 

And as I mentioned before, a statement in  README.md, CHANGELOG.md and CITATIONS.md, should clarify the relationship without a deep analysis of the two versions. If the users still need further details, we can add a separate changelog - "CHANGELOG_snakemake.md" to track the differences with the snakemake version. However, if the main CHANGELOG shows differences with the snakemake version, how do we track differences between Nextflow versions in the future.

Happy to have this conversation off GitHub and clarify my points.

Reply:
@alxndrdiaz, Updates from offline conversation: We are going to write all the documentation for Nextflow independently. Then we will create a new section in usage that will detail the differences between the Snakemake and Nextflow versions (maybe something similar to the dependencies table) and will also provide instructions for users moving from Snakemake. This section will be referenced in both the README and CHANGELOG.